gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALRasterSourceProvider.scala

```diff
@@ -17,17 +17,12 @@
 package geotrellis.contrib.vlm.gdal
 
 import geotrellis.contrib.vlm.RasterSourceProvider
-
+import geotrellis.contrib.vlm.avro.GeoTrellisDataPath
+import geotrellis.contrib.vlm.geotiff.GeoTiffDataPath
 
 class GDALRasterSourceProvider extends RasterSourceProvider {
   def canProcess(path: String): Boolean =
-    try {
-      GDALDataPath(path)
-      true
-    } catch {
-      case _: Throwable => false
-    }
+    (!path.startsWith(GeoTiffDataPath.PREFIX) && !path.startsWith(GeoTrellisDataPath.PREFIX)) && GDALDataPath.parseOption(path).nonEmpty && path.nonEmpty
```

Comment:
I thought that we've talked about it during paring, I think this logic `gt+`, `gdal+` and `gtiff+` it is only an SPI logic. Our parsers are agnostic to these prefix. 

Also `RasterSource(gtPath)` can return the actuall tiff, in case on the return side there would be a properly encoded GeoTiff with proper headers (yes, GDAL can read things that have weird URIs and the data extension as well as the data uri can be of different formats).

`gt+http://path/to/my/catalog?layer=name&zoom=10` can be also parsed by the `GDALRasterSource` properly.

Reply:
Yeah, you're right. Now that I'm thinking about it, we did discuss it. That makes sense :+1: